src/main/java/com/iexec/worker/sgx/SgxService.java

```diff
@@ -67,34 +99,45 @@ private boolean isSgxSupported() {
         return true;
     }
 
-    private boolean isSgxDevicePresent() {
+    private boolean isSgxDevicePresent(SgxDriverMode sgxDriverMode) {
         // "wallet-address-sgx-check" as containerName to avoid naming conflict
         // when running multiple workers on the same machine.
         String containerName = workerConfigService.getWorkerWalletAddress() + "-sgx-check";
         String alpineLatest = "alpine:latest";
-        String cmd = "find /dev -name isgx -exec echo true ;";
+
+        final String[] devices = sgxDriverMode.getDevices();
+        // Check all required devices exists
+        final String cmd = String.format("/bin/sh -c '%s; echo $?;'",
+                Arrays.stream(devices)
+                        .map(devicePath -> "test -e \"" + devicePath + "\"")
+                        .collect(Collectors.joining(" && ")));
 
         if (!dockerService.getClient().pullImage(alpineLatest)) {
             log.error("Failed to pull image for sgx check");
             return false;
         }
 
+        final List<String> devicesBind = Arrays.stream(devices)
+                .map(devicePath -> devicePath + ":" + devicePath)
+                .collect(Collectors.toList());
+
         DockerRunRequest dockerRunRequest = DockerRunRequest.builder()
                 .containerName(containerName)
                 .imageUri(alpineLatest)
                 .cmd(cmd)
                 .maxExecutionTime(60000) // 1 min
-                .binds(Collections.singletonList("/dev:/dev"))
+                .binds(devicesBind)
```

Comment:
I miss a point here.
In `DockerRunRequestBuilder#sgxDriverMode`, we use the `devices` methods, not the `binds` one.
Do both methods have the same behavior ?

Reply:
I don't find a lot of info on the difference between `devices` and `binds` but the first seems better. Let's update it!

Reply:
See https://github.com/iExecBlockchainComputing/iexec-worker/pull/434/commits/34aea8647a6918ffcb0c93c85c7362b016ebd805.